ios/brave-ios/Sources/BraveWallet/Crypto/Stores/TransactionConfirmationStore.swift

```diff
@@ -550,17 +550,21 @@ public class TransactionConfirmationStore: ObservableObject, WalletObserverStore
         gasSymbol = activeParsedTransaction.networkSymbol
         gasAssetRatio = assetRatios[activeParsedTransaction.networkSymbol.lowercased(), default: 0]
 
-        if let gasBalance = gasTokenBalanceCache[network.chainId]?[
+        let gasBalance = gasTokenBalanceCache[network.chainId]?[
           activeParsedTransaction.fromAccountInfo.id
-        ] {
-          if let gasValue = BDouble(gasFee.fee),
-            BDouble(gasBalance) > gasValue
-          {
-            isBalanceSufficient = true
+        ]
+        if let gasBalance,
+          let gasValue = BDouble(gasFee.fee),
+          let fromToken = details.fromToken,
+          let fromValue = BDouble(details.fromAmount)
+        {
+          if network.isNativeAsset(fromToken) {
```

Comment:
Do we want this native balance check for `.filSend` too? We get an error creating the tx but funds could be sent before tx approval